x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/lucene/ValuesSourceReaderOperator.java

```diff
@@ -241,9 +248,24 @@ private void loadFromManyLeaves(Block[] blocks, DocVector docVector) throws IOEx
         }
     }
 
-    private void trackStoredFields(StoredFieldsSpec spec) {
+    /**
+     * Is it more efficient to use a sequential stored field reader
+     * when reading stored fields for the documents contained in {@code docIds}.
+     */
+    private boolean isSequential(IntVector docIds) {
+        return docIds.getInt(docIds.getPositionCount() - 1) - docIds.getInt(0) == docIds.getPositionCount() - 1;
```

Comment:
I think we should only switch to the sequential mode when the number of documents exceeds a certain threshold? The search API uses 10 for this. The sequential (i.e., merging) stored field reader eagerly decompresses an entire block.

Reply:
I can do that! FWIW, we're not going to have many blocks less than 10, but it's fine to do.